### PR TITLE
Add Optuna study dashboard

### PR DESCRIPTION
**Pull request by tonypzy**
## Description

- `py/visdom/integrations/optuna.py`: Adds an optional Optuna study dashboard with summary, HParams, optimization history, timeline, parameter importance, multi-objective support, and periodic refresh.
- `README.md`: Documents the dashboard dependencies, configuration, and usage.

## Motivation and Context

The Optuna integration covers trial logging, study dashboards, browser integration, and intermediate reporting.
I plan to split the implementation into four smaller PRs:
- Optuna trial callback
- Study dashboard
- Browser integration
- Intermediate trial reporting

This is the second PR and only introduces the Study dashboard. The browser integration and intermediate trial reporting will be submitted separately after their dependencies are ready.

## How Has This Been Tested?

Run a simple case:
```
import time

import optuna
import visdom
from visdom.integrations import OptunaCallback


dashboard_env = "optuna_dashboard_demo_{}".format(int(time.time()))
viz = visdom.Visdom()

if not viz.check_connection(timeout_seconds=3):
    raise RuntimeError("Start the Visdom server before running this example.")

callback = OptunaCallback(
    viz,
    dashboard_env=dashboard_env,
    objective_names=["loss"],
    create_dashboard=True,
    refresh_every=5,
    raise_on_error=True,
)


def objective(trial):
    x = trial.suggest_float("x", -10, 10)
    y = trial.suggest_float("y", -5, 5)
    learning_rate = trial.suggest_float("learning_rate", 1e-4, 1e-1, log=True)
    optimizer = trial.suggest_categorical("optimizer", ["Adam", "SGD", "RMSprop"])

    optimizer_penalty = {"Adam": 0.0, "RMSprop": 0.5, "SGD": 1.0}[optimizer]
    time.sleep(0.05)
    return (
        (x - 2) ** 2
        + 0.5 * (y + 1) ** 2
        + (learning_rate - 0.01) ** 2 * 100
        + optimizer_penalty
    )


study = optuna.create_study(
    study_name="browser-dashboard-demo",
    direction="minimize",
    sampler=optuna.samplers.TPESampler(seed=42),
)
study.optimize(objective, n_trials=40, callbacks=[callback])
callback.update_dashboard(study)

print("\nBest value:", study.best_value)
print("Best parameters:", study.best_params)
print("Dashboard: http://localhost:8097/env/{}".format(dashboard_env))

```

## Screenshots (if appropriate):

https://github.com/user-attachments/assets/c2c19cc6-312e-4d4c-9a4b-0f64a04ae090

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactor or cleanup (changes to existing code for improved readability or performance)

## Checklist:

- [ ] I adapted the version number under `py/visdom/VERSION` according to [Semantic Versioning](https://semver.org/)
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Summary by Sourcery

Add optional Optuna study dashboards and intermediate-value visualization to the Visdom integration.

New Features:
- Add optional Optuna study dashboards with summaries, trial hyperparameters, optimization history, timelines, parameter importance, intermediate values, and multi-objective support.
- Enable automatic dashboard creation and periodic refresh alongside explicit final dashboard updates.
- Record Optuna intermediate trial values in Visdom trial environments.

Enhancements:
- Improve Optuna callback compatibility through metric-name fallbacks and reserved-tag handling.
- Add configurable dashboard error handling that can warn or raise without interrupting optimization by default.

Documentation:
- Document the Optuna dashboard setup, Plotly dependency, configuration, and usage.

Tests:
- Add unit coverage for Optuna compatibility and dashboard/trial error handling.

### CODE REVIEW

**Comment by tonypzy** [comment]
@vedansh-5 @rajnisht7 @Manik-Khajuria-5 @Jayantparashar10 Please review

**Review by Manik-Khajuria-5 — CHANGES_REQUESTED** [review]
@tonypzy raise on error may still let some errors escape because some operations happen outside the try block. I think it can be moved inside please check

**Review by Manik-Khajuria-5 — CHANGES_REQUESTED** [review]
@tonypzy The Optuna integration looks good overall. I found a couple of edge cases around older Optuna versions and the total tag limit that can cause all trials to fail silently. Could we handle these cases and add a few tests for them?

**Review by Manik-Khajuria-5 — COMMENTED** [review]
@tonypzy Can u please resolve the conflicts rest changes looks good to me

**Review by Manik-Khajuria-5 — APPROVED** [review]

**Review by rajnisht7 — APPROVED** [review]

**Review by Jayantparashar10 — APPROVED** [review]

**Review by vedansh-5 — APPROVED** [review]

**Review by Saksham-Sirohi — CHANGES_REQUESTED** [review]
please resolve conflicts

**Review by Saksham-Sirohi — APPROVED** [review]
